Add ConstantNaming analyzer

ddmler opened the pull request:
Resolves #125

K-Phoen commented on src/Analyzer/Pass/Statement/ConstantNaming.php:
```diff
@@ -0,0 +1,58 @@
+
+class ConstantNaming implements ConfigurablePassInterface, AnalyzerPassInterface
+{
+
```
Extra line here.

K-Phoen commented on src/Analyzer/Pass/Statement/ConstantNaming.php:
```diff
@@ -0,0 +1,58 @@
+                    $stmt
+                );
+
+                return true;
```
If you return inside the loop, all the constants might not be analyzed. Take the following code for instance, only the first invalid constant will be reported:

```php
class Foo {
    const Bar = 1, Baz = 2;
}
```

K-Phoen commented on src/Analyzer/Pass/Statement/ConstantNaming.php:
```diff
@@ -0,0 +1,58 @@
+    public function pass(ClassConst $stmt, Context $context)
+    {
+        foreach ($stmt->consts as $const) {
+            if ($const->name != strtoupper($const->name)) {
```
Why not using a strict equality? (`===`)

K-Phoen commented on src/Analyzer/Pass/Statement/ConstantNaming.php:
```diff
@@ -0,0 +1,58 @@
+    public function getRegister()
+    {
+        return [
+            \PhpParser\Node\Stmt\ClassConst::class
```
Unnecessary FQCN usage here: the ClassConst has been imported through a `use` statement.

K-Phoen commented on tests/analyze-fixtures/ConstantNaming.php:
```diff
@@ -0,0 +1,19 @@
+class ConstantNaming
+{
+    const UPPERCASE = 2;
+    const Lowercase = 1;
```
Could you add a case when several constants are declared using a single `const` keyword?

```php
const Foo = 1, Bar = 2;
```

ddmler commented:
@K-Phoen thanks for review I changed all your points.
I copied the FQCN from another analyzer so i changed it there too.
The return inside the loop would have been a minor problem since the notice would show the whole line and therefore both incorrect constants, but I changed it anyway.
Should we add the wrong constant name to the notice message?

K-Phoen commented:
> Should we add the wrong constant name to the notice message?

As you said, PHPSA shows the incriminated lines so that would not be really useful, wouldn't it? ¯\_(ツ)_/¯

ddmler commented:
ping @K-Phoen review issues are fixed can you check again?

K-Phoen commented:
LGTM :+1:
